functests/4_latency/latency.go

```diff
@@ -253,3 +245,49 @@ func getOslatPod(profile *performancev2.PerformanceProfile, node *corev1.Node) *
 		},
 	}
 }
+
+func createLatencyTestPod(testPod *corev1.Pod) {
+	err := testclient.Client.Create(context.TODO(), testPod)
+	Expect(err).ToNot(HaveOccurred())
+
+	timeout, err := strconv.Atoi(latencyTestRuntime)
+	Expect(err).ToNot(HaveOccurred())
+
+	By("Waiting two minutes to download the latencyTest image")
+	err = pods.WaitForPhase(testPod, corev1.PodRunning, 2*time.Minute)
+	Expect(err).ToNot(HaveOccurred())
+
+	By("Waiting another two minutes to give enough time for the cluster to move the pod to Succeeded phase")
+	podTimeout := time.Duration(timeout + 120)
+	err = pods.WaitForPhase(testPod, corev1.PodSucceeded, podTimeout*time.Second)
+	Expect(err).ToNot(HaveOccurred())
+}
+
+func extractLatencyValues(testName string, exp string, node *corev1.Node) string {
+	cmd := []string{"cat", fmt.Sprintf("/rootfs/var/log/%s.log", testName)}
+	out, err := nodes.ExecCommandOnNode(cmd, node)
+	Expect(err).ToNot(HaveOccurred())
+
+	maximumRegex, err := regexp.Compile(exp)
+	Expect(err).ToNot(HaveOccurred())
+
+	latencies := maximumRegex.FindStringSubmatch(out)[1]
+	Expect(maximumLatency).ToNot(BeNil())
+
+	return latencies
+}
+
+// initMaximumLatencyValue should look for one of the following environment variables:
+// OSLAT_MAXIMUM_LATENCY: the expected maximum latency for all buckets in us
+// CYCLICTEST_MAXIMUM_LATENCY: the expected maximum latency for all buckets in us
+// HWLATDETECT_MAXIMUM_LATENCY: the expected maximum latency for all buckets in us
+func initMaximumLatencyValue(testName string) {
+	envVariableName := fmt.Sprintf("%s_MAXIMUM_LATENCY", strings.ToUpper(testName))
```

Comment:
we should validate `testName` is one of the allowed values, return a proper error if not